### PR TITLE
depr(api): mark `ibis.sequence()` for removal

### PR DESCRIPTION
**Pull request by kszucs**
not required anymore, the API methods should automatically convert coerce to the right types

### CODE REVIEW

File: ibis/expr/api.py [hunk 1]
```diff
@@ -227,23 +225,6 @@ def param(type: dt.DataType) -> ir.Scalar:
     return ops.ScalarParameter(type).to_expr()
 
 
-# TODO(kszucs): should be deprecated
```
Comment (kszucs) [inline ibis/expr/api.py:230]: Shall we have a `4.2` release where we can deprecate this function and mark for removal in `5.0`?

Comment (cpcloud) [reply ibis/expr/api.py:230]: We can also deprecate in 5 and remove in 6.

Comment (kszucs) [reply ibis/expr/api.py:230]: Which one would you prefer?

Comment (cpcloud) [reply ibis/expr/api.py:230]: Let's deprecate in 5 and remove in 6.

Comment (kszucs) [reply ibis/expr/api.py:230]: Updated, the implementation was incorrect actually.

**Review by cpcloud — APPROVED** [review]
LGTM, thanks!

**Pull request merged**